### PR TITLE
4325 site struct

### PR DESCRIPTION
**Pull request by chiaberry**
# Description

Adding pages that dont have janisInstances (but do have janisUrls) to the site structure.

There are still pages that don't have instances or urls, They don't have departments nor topics assigned. They aren't being added to the site structure because they have no place in the hierarchy. (for example https://joplin-pr-import-everything.herokuapp.com/admin/pages/156/edit/#tab-content)

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 
     

https://janis-v3-4325-site-struct.netlify.app/

check the events at the bottom of the page. click one, confirm it loads. Click a location within that page, confirm it loads. 

# Checklist:
- [ ] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: src/js/helpers/cleanData.js [hunk 1]
```diff
@@ -288,8 +291,7 @@ export const cleanLinks = (links, pageType) => {
     const link = edge.node;
 
     // Check for global
-    //chia put this back, this is for local build
```
Comment (chiaberry) [inline src/js/helpers/cleanData.js:291]: 😬

File: src/js/helpers/cleanData.js [hunk 2]
```diff
@@ -89,8 +89,10 @@ export const cleanContact = contact => {
 };
 
 export const cleanLocationPageJanisUrl = janisUrl => {
-  // quick fix for urls until we get localized urls working in joplin
-  return `/${janisUrl.split('/en/')[1]}`;
+  if(!!janisUrl.length){
+    return janisUrl[0];
```
Comment (briaguya) [inline src/js/helpers/cleanData.js:93]: why is janisUrl an array here? not something that needs to be changed rn but might be nice to have a comment about

File: src/js/helpers/cleanData.js [hunk 3]
```diff
@@ -288,8 +291,7 @@ export const cleanLinks = (links, pageType) => {
     const link = edge.node;
```
Comment (briaguya) [inline src/js/helpers/cleanData.js:291]: are we still using all the pageType specific logic above this?

File: static.config.js [hunk 4]
```diff
@@ -735,6 +736,19 @@ const makeAllPages = async (langCode, incrementalPageId) => {
           ),
         ),
       );
+    }
+
+    // not all pages have instances (events and locations not under departments)
+    return Promise.all(
+      pageAtUrlInfo.node.janisUrls.map(instanceOfPage =>
```
Comment (briaguya) [inline static.config.js:743]: so are we not getting `janisUrls` on pages that have `janisInstances`? Are we building these pages twice and it's just working so we're cool with it? (i'm down)

File: src/js/helpers/cleanData.js (hunk 2 above)
Comment (chiaberry) [reply src/js/helpers/cleanData.js:93]: good question! the janisUrls that come back are the array of urls for the page, so its dept and topic pages instances, whereas before I believe it was just one url.

File: static.config.js (hunk 4 above)
Comment (chiaberry) [reply static.config.js:743]: we get janisUrls on the other pages, but they are built at their instances. event & location pages are always built at their janisUrl. 

I was going to say they aren't built twice, but now im thinking that they might be built twice at the same place if they have events and topics...

File: src/js/helpers/cleanData.js (hunk 3 above)
Comment (chiaberry) [reply src/js/helpers/cleanData.js:291]: the clean links pagetype logic?

Comment (briaguya) [reply src/js/helpers/cleanData.js:291]: yeah, we have stuff like 

```
  // Topics
  if (pageType === 'topic') {
    for (const edge of links.edges) {
      const link = edge.node;
      if (link.topiccollections && link.topiccollections.edges.length) {
        for (const edge of link.topiccollections.edges) {
          const { topiccollection } = edge.node;

          link.topLinks = [];
          link.otherLinks = [];
          link.url = `${pathPrefix || ''}/${link.slug}`;
          link.text = link.title;
          link.topiccollection = topiccollection;

          cleanedLinks.push(link);
        }
      }
    }

    return cleanedLinks;
  }
```

it seems like a good place to look for future cleanup

Comment (chiaberry) [reply src/js/helpers/cleanData.js:291]: ive cleaned up some, some of them are still being used. there are probably some i missed, or other refactoring we could do.

Comment (briaguya) [reply src/js/helpers/cleanData.js:291]: made an issue for future cleanup https://github.com/cityofaustin/techstack/issues/4330

**Pull request merged**